src/blockdigest.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "blockdigest.h"
+#include "streams.h"
+#include "version.h"
+#include "uint256.h"
+#include "primitives/transaction.h"
+#include "utiltime.h"
+#include "utilstrencodings.h"
+
+namespace DigestFilter {
+
+std::vector<uint8_t> base_filter::outpoint_data(const COutPoint& outpoint) const
+{
+    CDataStream stream(SER_NETWORK, PROTOCOL_VERSION);
+    stream << outpoint;
+    return std::vector<uint8_t>(stream.begin(), stream.end());
+}
+
+std::vector<const std::vector<uint8_t>> base_filter::tx_data_vec(const CTransaction& tx) const
+{
+    std::vector<const std::vector<uint8_t>> tx_data;
```

Comment:
I guess  I really don't understand what goes in the filter :p 

In any case, you can do: 

x.emplace_back()
x.back()

to construct in place and then get a reference to the last prevector to pass to outpoint_data :)

figuring out something to resize to would be good, but maybe it can't be done.